From a.beregalov@gmail.com  Fri Jul 10 15:22:21 2009
From: Alexander Beregalov <a.beregalov@gmail.com>
Date: Sat, 4 Jul 2009 23:09:32 +0400
Subject: Staging: vt665x: depend on WIRELESS_EXT
To: Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>
Cc: gregkh@suse.de, linux-kernel@vger.kernel.org, forest@alittletooquiet.net
Message-ID: <20090704190932.GA17284@orion>
Content-Disposition: inline

From: Alexander Beregalov <a.beregalov@gmail.com>

Make vt665x depended on WIRELESS_EXT and remove remain ifdef's

Signed-off-by: Alexander Beregalov <a.beregalov@gmail.com>
Cc: Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/vt6655/Kconfig       |    2 +-
 drivers/staging/vt6655/device.h      |    2 --
 drivers/staging/vt6655/device_main.c |    8 --------
 drivers/staging/vt6655/iwctl.c       |   10 ----------
 drivers/staging/vt6655/iwctl.h       |    4 ----
 drivers/staging/vt6656/Kconfig       |    1 +
 drivers/staging/vt6656/main_usb.c    |    2 --
 7 files changed, 2 insertions(+), 27 deletions(-)

--- a/drivers/staging/vt6655/device.h
+++ b/drivers/staging/vt6655/device.h
@@ -950,9 +950,7 @@ typedef struct __device_info {
     UINT                    uChannel;
     BOOL                    bMACSuspend;
 
-#ifdef WIRELESS_EXT
 	struct iw_statistics	wstats;		// wireless stats
-#endif /* WIRELESS_EXT */
     BOOL                    bCommit;
 
 } DEVICE_INFO, *PSDevice;
--- a/drivers/staging/vt6655/device_main.c
+++ b/drivers/staging/vt6655/device_main.c
@@ -1146,9 +1146,7 @@ device_found1(struct pci_dev *pcid, cons
     dev->irq                = pcid->irq;
     dev->netdev_ops         = &device_netdev_ops;
 
-#ifdef CONFIG_WIRELESS_EXT
 	dev->wireless_handlers = (struct iw_handler_def *)&iwctl_handler_def;
-#endif
 
     rc = register_netdev(dev);
     if (rc)
@@ -3396,10 +3394,8 @@ static int  device_ioctl(struct net_devi
 	PSDevice	        pDevice = (PSDevice)netdev_priv(dev);
 #endif
 
-#ifdef WIRELESS_EXT
 	struct iwreq *wrq = (struct iwreq *) rq;
 	int                 rc =0;
-#endif
     PSMgmtObject        pMgmt = pDevice->pMgmt;
     PSCmdRequest        pReq;
 
@@ -3411,8 +3407,6 @@ static int  device_ioctl(struct net_devi
 
     switch(cmd) {
 
-#ifdef WIRELESS_EXT
-
 	case SIOCGIWNAME:
 		rc = iwctl_giwname(dev, NULL, (char *)&(wrq->u.name), NULL);
 		break;
@@ -3751,8 +3745,6 @@ static int  device_ioctl(struct net_devi
 #endif // #ifdef WPA_SUPPLICANT_DRIVER_WEXT_SUPPORT
 //End Add -- //2008-0409-07, <Add> by Einsn Liu
 
-#endif // WIRELESS_EXT
-
     case IOCTL_CMD_TEST:
 
 		if (!(pDevice->flags & DEVICE_FLAGS_OPENED)) {
--- a/drivers/staging/vt6655/iwctl.c
+++ b/drivers/staging/vt6655/iwctl.c
@@ -80,8 +80,6 @@ extern WORD TxRate_iwconfig;//2008-5-8 <
 #define SUPPORTED_WIRELESS_EXT                  17
 #endif
 
-#ifdef WIRELESS_EXT
-
 static const long frequency_list[] = {
     2412, 2417, 2422, 2427, 2432, 2437, 2442, 2447, 2452, 2457, 2462, 2467, 2472, 2484,
     4915, 4920, 4925, 4935, 4940, 4945, 4960, 4980,
@@ -90,8 +88,6 @@ static const long frequency_list[] = {
     5700, 5745, 5765, 5785, 5805, 5825
 	};
 
-#endif
-
 
 /*---------------------  Static Classes  ----------------------------*/
 
@@ -105,8 +101,6 @@ static int          msglevel            
 
 /*---------------------  Export Variables  --------------------------*/
 
-#ifdef WIRELESS_EXT
-
 struct iw_statistics *iwctl_get_wireless_stats(struct net_device *dev)
 {
 	PSDevice pDevice = netdev_priv(dev);
@@ -2314,7 +2308,3 @@ const struct iw_handler_def	iwctl_handle
 	.private	= NULL,
 	.private_args	= NULL,
 };
-
-
-
-#endif // WIRELESS_EXT
--- a/drivers/staging/vt6655/iwctl.h
+++ b/drivers/staging/vt6655/iwctl.h
@@ -50,8 +50,6 @@ extern "C" {                            
 #endif /* __cplusplus */
 
 
-#ifdef WIRELESS_EXT
-
 struct iw_statistics *iwctl_get_wireless_stats (struct net_device *dev);
 
 
@@ -231,8 +229,6 @@ int iwctl_siwmlme(struct net_device *dev
 #endif // #ifdef WPA_SUPPLICANT_DRIVER_WEXT_SUPPORT
 
 
-#endif
-
 extern const struct iw_handler_def	iwctl_handler_def;
 extern const struct iw_priv_args	iwctl_private_args;
 
--- a/drivers/staging/vt6655/Kconfig
+++ b/drivers/staging/vt6655/Kconfig
@@ -1,6 +1,6 @@
 config VT6655
    tristate "VIA Technologies VT6655 support"
-   depends on PCI
+   depends on WIRELESS_EXT && PCI
    ---help---
    This is a vendor-written driver for VIA VT6655.
 
--- a/drivers/staging/vt6656/Kconfig
+++ b/drivers/staging/vt6656/Kconfig
@@ -1,5 +1,6 @@
 config VT6656
 	tristate "VIA Technologies VT6656 support"
+	depends on WIRELESS_EXT && USB
 	---help---
 	This is a vendor-written driver for VIA VT6656.
 
--- a/drivers/staging/vt6656/main_usb.c
+++ b/drivers/staging/vt6656/main_usb.c
@@ -856,9 +856,7 @@ vntwusb_found1(struct usb_interface *int
 
     netdev->netdev_ops         = &device_netdev_ops;
 
-#ifdef CONFIG_WIRELESS_EXT
 	netdev->wireless_handlers = (struct iw_handler_def *)&iwctl_handler_def;
-#endif /* WIRELESS_EXT */
 
    //2008-0623-01<Remark>by MikeLiu
   //2007-0821-01<Add>by MikeLiu
